predicators/structs.py

```diff
@@ -430,6 +430,12 @@ class Task:
     """Struct defining a task, which is an initial state and goal."""
     init: State
     goal: Set[GroundAtom]
+    # Sometimes we want the task presented to the agent to have goals described
+    # in terms of predicates that are different than those describing the goal
+    # of the task presented to the demonstrator. In these cases, we will store
+    # an "alternative goal" in this field and replace the goal with the
+    # alternative goal before giving the task to the agent.
+    alt_goal: Optional[Set[GroundAtom]] = field(default_factory=set)
```

Comment:
I think "alternative goal" is a decent name -- not great, but not terrible. Would be great if you have any suggestions for a better name.

Reply:
what about 'oracle goal', or 'demonstrator goal'?

Reply:
Well, the alternative goal is not for the oracle, it's for the agent. Calling it `agent_goal` is a possibility, but I think that is still confusing because then someone might think the normal goal is not for the agent. 

I think let's keep it `alt_goal` for now, and we can update it later if we want to.